test/helpers/dom.ts

```diff
@@ -0,0 +1,37 @@
+export const element = {
```

Comment:
All this should already be provided by jsdom (`document.createElement()`), with actual mock implementations?

Reply:
Yes, `document.createElement` is a dependency of solid-ui that is available in the browser as well as (via jsdom) in node. The same is true for rdflib and solid-auth-client, but I'm also mocking those, because none of them are part of the system under test, right? We discussed this on 6 Feb, and then we agreed to [mock anything that produces side effects](https://inrupt-team.slack.com/archives/CQKEE9G3W/p1580990593020800). Do you think we should revise that decision?

Reply:
Maybe in the unit tests we mock it, but in the integration tests we use jsdom?

Reply:
JSDOM already _is_ a mock of the DOM :) You should generally be able to assume that JSDOM will not have bugs - and the same holds true for rdflib and solid-auth-client. But you mock those because they make HTTP requests and you don't want to rely on some server being up and serving the right requests, and because you want to influence what they return. So you might want s-a-c to tell the system under test that a given WebID is logged in that you then make assertions on, or you instead that the user is logged out.

In the case of DOM elements, however, there is going to be no difference in the implementation - you always want `document.createElement` to return an element that e.g. has an `appendChild`.

Reply:
Using jsdom now.